HA/support-bundle-size-poc/ha-support-bundle-size-poc.md

```diff
@@ -0,0 +1,248 @@
+            -   Nov 27 16:26:59
+            -   2 node cluster
+        
+        2.  634KB pcsd.log
+
+
+<a id="org8aadcd5"></a>
+
+### <http://ssc-nfs-srvr2.pun.seagate.com/logs1/ldr-r1/EOS-15325/build532/secondary/SUPPORT_BUNDLE.SB1x0duu69o>
+
+1.  1M HA
+
+    ```
+    total 1.1M
+    1640262545 drwx------ 3 744354 744354   73 Feb  7 12:41 ha_SB1x0duu69
+    3233635359 -rw-r--r-- 1 744354 744354 1.1M Nov 28 21:10 ha_SB1x0duu69.tar.gz
+    
+    ./ha_SB1x0duu69:
+    total 19M
+    2162752323 drwxrwxr-x 2 744354 744354  141 Nov 28 21:10 ha
+    1640262548 -rw-rw---- 1 744354 744354  19M Nov 28 21:10 corosync.log
+    1640262547 -rw-rw---- 1 744354 744354    0 Nov 28 12:01 pacemaker.log
+    1640262546 -rw------- 1 744354 744354 121K Nov 28 21:05 pcsd.log
+    
+    ./ha_SB1x0duu69/ha:
+    total 196K
+    2162752324 -rw-r--r-- 1 744354 744354  22K Nov 28 21:10 cortxha.log
+    2162752325 -rw-r--r-- 1 744354 744354 5.1K Nov 28 21:10 ha_cmds_output
+    2162752329 -rw-rwxr-- 1 744354 744354 2.1K Nov 28 10:02 ha_setup.log
+    2162752328 -rw-r----- 1 744354 744354 132K Nov 28 21:01 pcmk_alert.log
+    2162752327 -rw-r----- 1 744354 744354  27K Nov 28 20:33 resource_agent.log
+    2162752330 -rw-r--r-- 1 744354 744354    0 Nov 28 21:10 support_bundle.err
+    ```
+    
+    1.  19M corosync.log
+    
+        -   Nov 29 00:31:07
+        -   Nov 29 09:40:36
+        -   2 node cluster
+        
+        1.  huge amount of fence\_ipmilan mesages (enabled recently to have information about fencing agent behavior)
+        
+            Logging configuration shall be revised or logs shall be disabled  
+
+
+<a id="orgc0ba216"></a>
+
+### <http://ssc-nfs-srvr2.pun.seagate.com/logs1/ldr-r1/certification/Node-1/SUPPORT_BUNDLE.SBeiyo8xuq>
+
+1GB  
+
+1.  28K HA
+
+    Doesn't make sense to check  
+
+
+<a id="org2546d83"></a>
+
+### <http://ssc-nfs-srvr2.pun.seagate.com/logs1/ldr-r1/EOS-16253/live_node/SUPPORT_BUNDLE.SBsifj5q2r;925553016>
+
+900M  
+
+1.  
+
+
+<a id="org465ec25"></a>
+
+## corosync.log size reduction options
+
+Since that file consumes major part of disk space in support bunle, lets check which options do we have to optimize the size.  
+
+
+<a id="org1c78c99"></a>
+
+### Initial number of lines: 127K, Size: 19M
+
+
+<a id="org6399097"></a>
+
+### sed -E /fence\_ipmilan.\*stderr/d gives 127K -> 94K: -5MiB
+
+This can be disabled in fence\_ipmilan configuration for sure.  
+
+
+<a id="org2f3dcbb"></a>
+
+### sed -E /common\_print/d gives 94K -> 76K: -3MiB
+
+Need to research whether common\_prints caused by \`pcs status\` invocations can be disabled via pacemaker configuration  
+
+
+<a id="orgd0f8eca"></a>
+
+### sed -E /cib:/d gives 76K -> 58K: -3.2MiB
+
+CIB updates is not really interesing information for analysis, but I would not recommend to remove it from logs  
+
+
+<a id="org61b6731"></a>
+
+### Generic solution: reduce size of logs in logrotate
+
+
+<a id="org83a2bf1"></a>
+
+### Pacemaker logging options

```

Comment:
Can we categorize logs in different categories like
1. corosync/pacemaker
2. Cortx-ha module logs

Then we can filter logs using these types and also add duration like number of days.
Can we come up with such scheme to filter logs whenever required in large cluster?

Reply:
I didn't understand where it is suggested to categorize them.
HA logs were excluded from analysis due to their minimal impact on support bundle size.  We probably can safely specify a month for their rotation. The trick is that we don't have LR2 components implemented yet. Their logs size are expected to be much bigger than for LR1 release i presume.
If the idea is to separate pacemaker logs from HA component logs due to 3rd party nature of pacemaker component - then, i would agree with that approach. It definitely makes sense.

Reply:
Yes. The log size would be more in R2. I was thinking, if the problem is in Cortx module then it doesn't make sense to collect corosync/pacemaker logs and vice versa. So, a filter would reduce size of logs further. Also a time filter would be helpful to collect logs of relevant time period if the time is known when issue occurred.

I see logrotate is configured for HA logs. We should implement it for corosync/pacemaker as well.

Reply:
Updated summary with an option to implement advanced filtering logic for HA support bundle script.

Reply:
Summary has generic statement, no concrete suggestions. I am merging it anyways, we are agile and think about while implementing.